scenarios/aws/microVMs/microvms/run.go

```diff
@@ -90,14 +90,11 @@ func newEC2Instance(awsEnv aws.Environment, name, ami, arch, instanceType, keyPa
 	return instance, err
 }
 
-func newMetalInstance(e commonConfig.CommonEnvironment, name, arch string, m config.DDMicroVMConfig) (*Instance, error) {
+func newMetalInstance(instanceEnv *InstanceEnvironment, name, arch string, m config.DDMicroVMConfig) (*Instance, error) {
 	var instanceType string
 	var ami string
 
-	awsEnv, err := aws.NewEnvironment(e.Ctx, aws.WithCommonEnvironment(&e))
-	if err != nil {
-		return nil, err
-	}
+	awsEnv := instanceEnv.Environment
```

Comment:
should we check the pointer for nil?

Reply:
Same as below. We should just panic here.